packages/core/src/lib/base_protocol.ts

```diff
@@ -16,27 +16,22 @@ import {
 import { filterPeersByDiscovery } from "./filterPeers.js";
 import { StreamManager } from "./stream_manager.js";
 
-const DEFAULT_NUM_PEERS_TO_USE = 3;
-
 /**
  * A class with predefined helpers, to be used as a base to implement Waku
  * Protocols.
  */
-export class BaseProtocol implements IBaseProtocol {
+export class BaseProtocol implements IBaseProtocolCore {
```

Comment:
Why these rename? I think we can keep previous names, no need to add `core` as it is already clear from the package name.

Reply:
I refactor these names as we now have two types of `BaseProtocol`:
- one on the protocol implementation level (in `@waku/core`)
- other on the SDK level (in `@waku/sdk`)

I personally incline towards having more verbose & readable names.